extensions/amp-list/0.1/amp-list.js

```diff
@@ -45,6 +44,7 @@ import {isArray, toArray} from '../../../src/types';
 import {isExperimentOn} from '../../../src/experiments';
 import {px, setStyles, toggle} from '../../../src/style';
 import {removeChildren} from '../../../src/dom';
+import setDOM from 'set-dom/dist/set-dom.js';
```

Comment:
I'm surprised this works, because this is definitely not an ESM.

Reply:
I came across https://github.com/DylanPiercey/set-dom/issues/1, which led me to try importing `set-dom/dist/set-dom.js`. I think the operative part is that it no longer needs a transformation from common JS.

Let me know if you have any concerns with merging this as is, given that all tests passed, and I've managed to verify the manual tests in `examples/` for `amp-list`.

Reply:
Added @choumx to review the `set-dom` changes in this PR.

Reply:
Can you paste the dist for amp-list?

Reply:
Here's the diff for `dist/v0/amp-list-0.1.js` before and after this PR. Full files are [here](https://gist.github.com/rsimha/ac93bb35c0b22b6ee7fcb9a9419feba4).
```diff
diff --git a/before.js b/after.js
index 2d0b41e..18b4271 100644
--- a/before.js
+++ b/after.js
@@ -1,67 +1,67 @@
-(self.AMP=self.AMP||[]).push({n:"amp-list",v:"1907012128380",f:(function(AMP,_){
-var m,aa="function"==typeof Object.create?Object.create:function(a){function b(){}b.prototype=a;return new b},n;if("function"==typeof Object.setPrototypeOf)n=Object.setPrototypeOf;else{var p;a:{var ba={a:!0},ca={};try{ca.__proto__=ba;p=ca.a;break a}catch(a){}p=!1}n=p?function(a,b){a.__proto__=b;if(a.__proto__!==b)throw new TypeError(a+" is not extensible");return a}:null}var da=n,q=window.DOMParser&&new window.DOMParser,ea=!1,fa=!1;
-try{var ha=q.parseFromString('<wbr class="A"/>',"text/html").body.firstChild,ia=document.createElement("div");ia.appendChild(ha);if("A"!==ia.firstChild.classList[0])throw Error();ea=!0}catch(a){}var ja=document.implementation.createHTMLDocument(""),r=ja.documentElement,t=ja.body;try{r.innerHTML+="",fa=!0}catch(a){q.parseFromString('<wbr class="A"/>',"application/xhtml+xml");var ka=/(<body[^>]*>)([\s\S]*)<\/body>/}
-function la(a,b){var c=q.parseFromString(a,"text/html");return c.body?"HTML"===b?c.documentElement:c.body.firstChild:ma(a,b)}function ma(a,b){if("HTML"===b){if(fa)return r.innerHTML=a,r;var c=a.match(ka);if(c){var d=c[2],e=c.index+c[1].length,f=e+d.length;a=a.slice(0,e)+a.slice(f);t.innerHTML=d}a=q.parseFromString(a,"application/xhtml+xml");for(b=a.body;t.firstChild;)b.appendChild(t.firstChild);return a.documentElement}t.innerHTML=a;return t.firstChild}var na=ea?la:ma;var w={default:function(a,b){if(!a||!a.nodeType)throw Error("set-dom: You must provide a valid node to update.");a.nodeType===oa&&(a=a.documentElement);b.nodeType===pa?qa(a,b):u(a,"string"===typeof b?na(b,a.nodeName):b);a[ra]||(a[ra]=!0,v(a))}};w.default.KEY="data-key";w.default.IGNORE="data-ignore";w.default.CHECKSUM="data-checksum";var sa="_set-dom-",ra=sa+"mounted",ta=1,oa=9,pa=11;
-function u(a,b){if(a.nodeType===b.nodeType)if(a.nodeType===ta){if(!(null!=a.getAttribute(w.default.IGNORE)&&null!=b.getAttribute(w.default.IGNORE)||(a.getAttribute(w.default.CHECKSUM)||NaN)===(b.getAttribute(w.default.CHECKSUM)||NaN)||a.isEqualNode(b)))if(qa(a,b),a.nodeName===b.nodeName){a=a.attributes;b=b.attributes;var c,d;for(c=a.length;c--;){var e=a[c];var f=e.namespaceURI;var g=e.localName;(d=b.getNamedItemNS(f,g))||a.removeNamedItemNS(f,g)}for(c=b.length;c--;)(e=b[c],f=e.namespaceURI,g=e.localName,
-d=a.getNamedItemNS(f,g),d)?d.value!==e.value&&(d.value=e.value):(b.removeNamedItemNS(f,g),a.setNamedItemNS(e))}else{for(var h=b.cloneNode();a.firstChild;)h.appendChild(a.firstChild);a.parentNode.replaceChild(h,a)}}else a.nodeValue!==b.nodeValue&&(a.nodeValue=b.nodeValue);else a.parentNode.replaceChild(b,x(a,"dismount")),v(b)}
-function qa(a,b){var c,d,e,f=a.firstChild;b=b.firstChild;for(var g=0;f;){g++;var h=f;var k=y(h);f=f.nextSibling;k&&(e||(e={}),e[k]=h)}for(f=a.firstChild;b;){g--;var l=b;b=b.nextSibling;e&&(c=y(l))&&(d=e[c])?(delete e[c],d!==f?a.insertBefore(d,f):f=f.nextSibling,u(d,l)):f?(h=f,f=f.nextSibling,y(h)?(a.insertBefore(l,h),v(l)):u(h,l)):(a.appendChild(l),v(l))}for(k in e)g--,a.removeChild(x(e[k],"dismount"));for(;0<=--g;)a.removeChild(x(a.lastChild,"dismount"))}
-function y(a){if(a.nodeType===ta&&(a=a.getAttribute(w.default.KEY)||a.id))return sa+a}function v(a){x(a,"mount")}function x(a,b){if(y(a)){var c=document.createEvent("Event"),d={value:a};c.initEvent(b,!1,!1);Object.defineProperty(c,"target",d);Object.defineProperty(c,"srcElement",d);a.dispatchEvent(c)}for(var e=a.firstChild;e;)e=x(e,b).nextSibling;return a};function ua(){var a,b;this.promise=new Promise(function(c,d){a=c;b=d});this.resolve=a;this.reject=b};function z(a,b){b=void 0===b?"":b;try{return decodeURIComponent(a)}catch(c){return b}};var va=/(?:^[#?]?|&)([^=&]+)(?:=([^&]*))?/g;function A(a){var b=Object.create(null);if(!a)return b;for(var c;c=va.exec(a);){var d=z(c[1],c[1]),e=c[2]?z(c[2],c[2]):"";b[d]=e}return b};var wa="";
-function xa(){var a,b=a||self;if(b.AMP_MODE)var c=b.AMP_MODE;else{c=b;var d=self.AMP_CONFIG||{},e=!!d.test||!(!c.AMP_TEST&&!c.__karma__),f=c.__karma__&&c.__karma__.config.amp.testOnIe,g=!!d.localDev||e,h=A(c.location.originalHash||c.location.hash);d=d.spt;var k=A(c.location.search);wa||(wa=g?"1907012128380":c.AMP_CONFIG&&c.AMP_CONFIG.v?c.AMP_CONFIG.v:"011907012128380");c=b.AMP_MODE={localDev:g,development:!("1"!=h.development&&!c.AMP_DEV_MODE),examiner:"2"==h.development,filter:h.filter,geoOverride:h["amp-geo"],
-userLocationOverride:h["amp-user-location"],minified:!0,lite:void 0!=k.amp_lite,test:e,testIe:f,log:h.log,version:"1907012128380",rtvVersion:wa,singlePassType:d}}return c};var ya=Object.prototype.toString;function B(a){return Array.isArray(a)};self.log=self.log||{user:null,dev:null,userForEmbed:null};var C=self.log;function E(){if(!C.user)throw Error("failed to call initLogConstructor");return C.user}function za(){if(C.dev)return C.dev;throw Error("failed to call initLogConstructor");}function F(a,b,c,d,e){E().assert(a,b,c,d,e,void 0,void 0,void 0,void 0,void 0,void 0)};function G(a){return a||{}};var H;function Aa(a){var b=a.ownerDocument||a;H&&H.ownerDocument===b||(H=b.createElement("div"));return Ba}function Ba(a){var b=H;b.innerHTML=a[0];a=b.firstElementChild;b.removeChild(a);return a};var I,Ca="Webkit webkit Moz moz ms O o".split(" ");function J(a,b){for(var c in b){var d=a,e=b[c];var f=d.style;var g=c;if(2>g.length?0:0==g.lastIndexOf("--",0))f=g;else{I||(I=Object.create(null));var h=I[g];if(!h){h=g;if(void 0===f[g]){var k=g;k=k.charAt(0).toUpperCase()+k.slice(1);b:{for(var l=0;l<Ca.length;l++){var D=Ca[l]+k;if(void 0!==f[D]){k=D;break b}}k=""}void 0!==f[k]&&(h=k)}I[g]=h}f=h}f&&(d.style[f]=e)}}
-function Da(a){var b=!1;void 0===b&&(b=a.hasAttribute("hidden"));b?a.removeAttribute("hidden"):a.setAttribute("hidden","")};var Ea=['\n      <i-amphtml-sizer class="i-amphtml-sizer">\n        <img alt="" role="presentation" aria-hidden="true"\n             class="i-amphtml-intrinsic-sizer" />\n      </i-amphtml-sizer>'];Ea.raw=Ea.slice();/*
+(self.AMP=self.AMP||[]).push({n:"amp-list",v:"1907012221080",f:(function(AMP,_){
+var n,aa="function"==typeof Object.create?Object.create:function(a){function b(){}b.prototype=a;return new b},q;if("function"==typeof Object.setPrototypeOf)q=Object.setPrototypeOf;else{var t;a:{var ba={a:!0},ca={};try{ca.__proto__=ba;t=ca.a;break a}catch(a){}t=!1}q=t?function(a,b){a.__proto__=b;if(a.__proto__!==b)throw new TypeError(a+" is not extensible");return a}:null}var da=q;function ea(){var a,b;this.promise=new Promise(function(c,d){a=c;b=d});this.resolve=a;this.reject=b};function u(a,b){b=void 0===b?"":b;try{return decodeURIComponent(a)}catch(c){return b}};var fa=/(?:^[#?]?|&)([^=&]+)(?:=([^&]*))?/g;function v(a){var b=Object.create(null);if(!a)return b;for(var c;c=fa.exec(a);){var d=u(c[1],c[1]),e=c[2]?u(c[2],c[2]):"";b[d]=e}return b};var w="";
+function ha(){var a,b=a||self;if(b.AMP_MODE)var c=b.AMP_MODE;else{c=b;var d=self.AMP_CONFIG||{},e=!!d.test||!(!c.AMP_TEST&&!c.__karma__),g=c.__karma__&&c.__karma__.config.amp.testOnIe,h=!!d.localDev||e,f=v(c.location.originalHash||c.location.hash);d=d.spt;var k=v(c.location.search);w||(w=h?"1907012221080":c.AMP_CONFIG&&c.AMP_CONFIG.v?c.AMP_CONFIG.v:"011907012221080");c=b.AMP_MODE={localDev:h,development:!("1"!=f.development&&!c.AMP_DEV_MODE),examiner:"2"==f.development,filter:f.filter,geoOverride:f["amp-geo"],userLocationOverride:f["amp-user-location"],
+minified:!0,lite:void 0!=k.amp_lite,test:e,testIe:g,log:f.log,version:"1907012221080",rtvVersion:w,singlePassType:d}}return c};var ia=Object.prototype.toString;function x(a){return Array.isArray(a)};self.log=self.log||{user:null,dev:null,userForEmbed:null};var z=self.log;function A(){if(!z.user)throw Error("failed to call initLogConstructor");return z.user}function B(){if(z.dev)return z.dev;throw Error("failed to call initLogConstructor");}function C(a,b,c,d,e){A().assert(a,b,c,d,e,void 0,void 0,void 0,void 0,void 0,void 0)};function D(a){return a||{}};var E;function F(a){var b=a.ownerDocument||a;E&&E.ownerDocument===b||(E=b.createElement("div"));return ja}function ja(a){var b=E;b.innerHTML=a[0];a=b.firstElementChild;b.removeChild(a);return a};var G,ka="Webkit webkit Moz moz ms O o".split(" ");function H(a,b){for(var c in b){var d=a,e=b[c];var g=d.style;var h=c;if(2>h.length?0:0==h.lastIndexOf("--",0))g=h;else{G||(G=Object.create(null));var f=G[h];if(!f){f=h;if(void 0===g[h]){var k=h;k=k.charAt(0).toUpperCase()+k.slice(1);b:{for(var l=0;l<ka.length;l++){var r=ka[l]+k;if(void 0!==g[r]){k=r;break b}}k=""}void 0!==g[k]&&(f=k)}G[h]=f}g=f}g&&(d.style[g]=e)}}
+function la(a){var b=!1;void 0===b&&(b=a.hasAttribute("hidden"));b?a.removeAttribute("hidden"):a.setAttribute("hidden","")};var ma=['\n      <i-amphtml-sizer class="i-amphtml-sizer">\n        <img alt="" role="presentation" aria-hidden="true"\n             class="i-amphtml-intrinsic-sizer" />\n      </i-amphtml-sizer>'];ma.raw=ma.slice();/*
  https://mths.be/cssescape v1.5.1 by @mathias | MIT license */
-var Fa;function Ga(a,b,c){if(b(a))c();else{var d=a.ownerDocument.defaultView;if(d.MutationObserver){var e=new d.MutationObserver(function(){b(a)&&(e.disconnect(),c())});e.observe(a,{childList:!0})}else var f=d.setInterval(function(){b(a)&&(d.clearInterval(f),c())},5)}}function Ha(a,b){Ga(a.documentElement,function(){return!!a.body},b)}function Ia(a){return new Promise(function(b){return Ha(a,b)})}function Ja(a){for(;a.firstChild;)a.removeChild(a.firstChild)}
-function K(a,b){/^[\w-]+$/.test(b);b="> ["+b+"]";if(void 0!==Fa)var c=Fa;else{try{var d=a.ownerDocument,e=d.createElement("div"),f=d.createElement("div");e.appendChild(f);c=e.querySelector(":scope div")===f}catch(g){c=!1}c=Fa=c}c?a=a.querySelector(b.replace(/^|,/g,"$&:scope ")):(a.classList.add("i-amphtml-scoped"),b=b.replace(/^|,/g,"$&.i-amphtml-scoped "),b=a.querySelectorAll(b),a.classList.remove("i-amphtml-scoped"),a=void 0===b[0]?null:b[0]);return a};var Ka=['\n        <amp-list-load-more load-more-failed class="i-amphtml-default-ui">\n          <div class="i-amphtml-list-load-more-message">\n            Unable to Load More\n          </div>\n          <button load-more-clickable\n            class="i-amphtml-list-load-more-button\n                  i-amphtml-list-load-more-button-has-icon\n                  i-amphtml-list-load-more-button-small"\n          >\n            <div class="i-amphtml-list-load-more-icon"></div>\n            <label>Retry</label>\n          </button>\n        </amp-list-load-more>\n      '];
-Ka.raw=Ka.slice();var La=['\n        <amp-list-load-more load-more-loading class="i-amphtml-default-ui">\n          <div class="i-amphtml-list-load-more-spinner"></div>\n        </amp-list-load-more>\n      '];La.raw=La.slice();var Ma=['\n        <amp-list-load-more load-more-button\n          class="amp-visible i-amphtml-default-ui">\n          <button load-more-clickable class="i-amphtml-list-load-more-button">\n            <label>See More</label>\n          </button>\n        </amp-list-load-more>\n      '];
-Ma.raw=Ma.slice();function Na(a){this.j=a;this.G=this.N=this.C=this.D=this.M=this.l=null}m=Na.prototype;m.initializeLoadMore=function(){Oa(this);Pa(this);Qa(this);this.G||(this.G=K(this.j,"load-more-end"))&&this.j.appendChild(this.G)};function Oa(a){a.l=K(a.j,"load-more-button");a.l?a.l.classList.add("amp-visible"):a.l=Aa(a.j)(Ma);a.j.appendChild(a.l);J(a.l,{visibility:"hidden"})}function Pa(a){a.D=K(a.j,"load-more-loading");a.D||(a.D=Aa(a.j)(La));a.j.appendChild(a.D)}
-m.getLoadMoreButton=function(){this.l||Oa(this);return this.l};m.getLoadMoreLoadingElement=function(){this.D||Pa(this);return this.D};m.getLoadMoreButtonClickable=function(){if(!this.M){var a=this.getLoadMoreButton();this.M=K(a,"load-more-clickable")||a}return this.M};function Qa(a){a.C=K(a.j,"load-more-failed");a.C||(a.C=Aa(a.j)(Ka));a.j.appendChild(a.C)}m.getLoadMoreFailedElement=function(){this.C||Qa(this);return this.C};
-m.getLoadMoreFailedClickable=function(){if(!this.N){var a=this.getLoadMoreFailedElement();this.N=K(a,"load-more-clickable")||a}return this.N};m.getLoadMoreEndElement=function(){return this.G};m.setLoadMoreEnded=function(){this.getLoadMoreFailedElement().classList.toggle("amp-visible",!1);this.getLoadMoreButton().classList.toggle("amp-visible",!1);this.getLoadMoreLoadingElement().classList.toggle("amp-visible",!1);var a=this.getLoadMoreEndElement();a&&a.classList.toggle("amp-visible",!0)};
-m.toggleLoadMoreLoading=function(a){a&&this.getLoadMoreFailedElement().classList.toggle("amp-visible",!1);var b=this.getLoadMoreEndElement();b&&b.classList.toggle("amp-visible",!1);this.getLoadMoreButton().classList.toggle("amp-visible",!a);this.getLoadMoreLoadingElement().classList.toggle("amp-visible",a)};
-m.setLoadMoreFailed=function(){var a=this.getLoadMoreFailedElement(),b=this.getLoadMoreButton();if(a||b)a.classList.toggle("amp-visible",!0),b.classList.toggle("amp-visible",!1),this.getLoadMoreLoadingElement().classList.toggle("amp-visible",!1)};function Ra(a,b){var c=a.ownerDocument.defaultView,d=c!=(c.__AMP_TOP||c);d?b=Sa(c,b)?L(c,b):null:(a=M(a),a=N(a),b=Sa(a,b)?L(a,b):null);return b}function O(a,b){a=a.__AMP_TOP||a;return L(a,b)}function Ta(a){var b=M(a);b=N(b);return L(b,"viewer")}function Ua(a,b){return Va(N(a),b)}function M(a){return a.nodeType?O((a.ownerDocument||a).defaultView,"ampdoc").getAmpDoc(a):a}function N(a){a=M(a);return a.isSingleDoc()?a.win:a}
-function L(a,b){Sa(a,b);var c=Wa(a);a=c[b];a.obj||(a.obj=new a.ctor(a.context),a.ctor=null,a.context=null,a.resolve&&a.resolve(a.obj));return a.obj}function Xa(a,b){var c=Va(a,b);if(c)return c;a=Wa(a);a[b]=Ya();return a[b].promise}function Va(a,b){var c=Wa(a)[b];if(c){if(c.promise)return c.promise;L(a,b);return c.promise=Promise.resolve(c.obj)}return null}function Wa(a){var b=a.services;b||(b=a.services={});return b}function Sa(a,b){a=a.services&&a.services[b];return!(!a||!a.ctor&&!a.obj)}
-function Ya(){var a=new ua,b=a.promise,c=a.resolve;a=a.reject;b.catch(function(){});return{obj:null,promise:b,resolve:c,reject:a,context:null,ctor:null}};function Za(a,b,c){var d=Ua(a,b);if(d)return d;var e=M(a);return e.waitForBodyOpen().then(function(){return $a(e.win,c,e.win.document.head)}).then(function(){var d=e.win;return d.ampExtendedElements&&d.ampExtendedElements[c]?Xa(N(a),b):null})}function ab(a){var b=Ra(a,"bind");if(b)return Promise.resolve(b);b=a.ownerDocument.defaultView;return b!==(b.__AMP_TOP||b)?bb(b):Za(a,"bind","amp-bind")}
-function $a(a,b,c){if(c){var d=[];c=c.querySelectorAll("script[custom-element]");for(var e=0;e<c.length;e++)d.push(c[e].getAttribute("custom-element"))}else d=[];if(!d.includes(b))return Promise.resolve();var f=O(a,"extensions");return f.waitForExtension(a,b)}function bb(a){return Ia(a.document).then(function(){return $a(a,"amp-bind",a.document.head)}).then(function(){return a.ampExtendedElements&&a.ampExtendedElements["amp-bind"]?Xa(a,"bind"):null})};function P(a,b){var c,d=this;this.Z=O(a,"timer");this.da=b;this.ca=c||0;this.H=-1;this.P=0;this.S=!1;this.aa=function(){d.H=-1;d.P=0;d.S=!0;d.da();d.S=!1}}P.prototype.isPending=function(){return-1!=this.H};P.prototype.schedule=function(a){var b=a||this.ca;this.S&&10>b&&(b=10);var c=Date.now()+b;return!this.isPending()||-10>c-this.P?(this.cancel(),this.P=c,this.H=this.Z.delay(this.aa,b),!0):!1};P.prototype.cancel=function(){this.isPending()&&(this.Z.cancel(this.H),this.H=-1)};function Q(){var a=100;this.ba=a;this.K=this.T=0;this.J=Object.create(null)}Q.prototype.has=function(a){return!!this.J[a]};Q.prototype.get=function(a){var b=this.J[a];if(b)return b.access=++this.K,b.payload};Q.prototype.put=function(a,b){this.has(a)||this.T++;this.J[a]={payload:b,access:this.K};if(!(this.T<=this.ba)){za().warn("lru-cache","Trimming LRU cache");a=this.J;var c=this.K+1,d;for(d in a){var e=a[d].access;if(e<c){c=e;var f=d}}void 0!==f&&(delete a[f],this.T--)}};var R=self.AMP_CONFIG||{},cb={thirdParty:R.thirdPartyUrl||"https://3p.ampproject.net",thirdPartyFrameHost:R.thirdPartyFrameHost||"ampproject.net",thirdPartyFrameRegex:("string"==typeof R.thirdPartyFrameRegex?new RegExp(R.thirdPartyFrameRegex):R.thirdPartyFrameRegex)||/^d-\d+\.ampproject\.net$/,cdn:R.cdnUrl||"https://cdn.ampproject.org",cdnProxyRegex:("string"==typeof R.cdnProxyRegex?new RegExp(R.cdnProxyRegex):R.cdnProxyRegex)||/^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org$/,localhostRegex:/^https?:\/\/localhost(:\d+)?$/,
-errorReporting:R.errorReportingUrl||"https://amp-error-reporting.appspot.com/r",localDev:R.localDev||!1,trustedViewerHosts:[/(^|\.)google\.(com?|[a-z]{2}|com?\.[a-z]{2}|cat)$/,/(^|\.)gmail\.dev$/]};var db=G({c:!0,v:!0,a:!0,ad:!0,action:!0}),eb,fb,hb=/[?&]amp_js[^&]*/,ib=/[?&]amp_gsa[^&]*/,jb=/[?&]amp_r[^&]*/,kb=/[?&]amp_kit[^&]*/,lb=/[?&]usqp[^&]*/;
-function S(a){var b;eb||(eb=self.document.createElement("a"),fb=self.UrlCache||(self.UrlCache=new Q));var c=b?null:fb,d=eb;if(c&&c.has(a))a=c.get(a);else{d.href=a;d.protocol||(d.href=d.href);var e={href:d.href,protocol:d.protocol,host:d.host,hostname:d.hostname,port:"0"==d.port?"":d.port,pathname:d.pathname,search:d.search,hash:d.hash,origin:null};"/"!==e.pathname[0]&&(e.pathname="/"+e.pathname);if("http:"==e.protocol&&80==e.port||"https:"==e.protocol&&443==e.port)e.port="",e.host=e.hostname;e.origin=
-d.origin&&"null"!=d.origin?d.origin:"data:"!=e.protocol&&e.host?e.protocol+"//"+e.host:e.href;d=xa().test&&Object.freeze?Object.freeze(e):e;c&&c.put(a,d);a=d}return a}function mb(a){var b=[],c;for(c in a){var d=a[c];if(null!=d)if(B(d))for(var e=0;e<d.length;e++){var f=d[e];b.push(encodeURIComponent(c)+"="+encodeURIComponent(f))}else e=d,b.push(encodeURIComponent(c)+"="+encodeURIComponent(e))}return b.join("&")}
-function nb(a,b){var c=void 0===c?"source":c;F(null!=a,"%s %s must be available",b,c);var d=a;a=d;"string"==typeof a&&(a=S(a));var e;(e="https:"==a.protocol||"localhost"==a.hostname||"127.0.0.1"==a.hostname)||(a=a.hostname,e=a.length-10,e=0<=e&&a.indexOf(".localhost",e)==e);F(e||/^(\/\/)/.test(d),'%s %s must start with "https://" or "//" or be relative and served from either https or from localhost. Invalid value: %s',b,c,d)}
-function ob(a){"string"==typeof a&&(a=S(a));var b=a;"string"==typeof b&&(b=S(b));if(!cb.cdnProxyRegex.test(b.origin))return a.href;b=a.pathname.split("/");F(db[b[1]],"Unknown path prefix in url %s",a.href);var c=b[2],d="s"==c?"https://"+decodeURIComponent(b[3]):"http://"+decodeURIComponent(c);F(0<d.indexOf("."),"Expected a . in origin %s",d);b.splice(1,"s"==c?3:2);b=d+b.join("/");d=(d=a.search)&&"?"!=d?(d=d.replace(hb,"").replace(ib,"").replace(jb,"").replace(kb,"").replace(lb,"").replace(/^[?&]/,
-""))?"?"+d:"":"";return b+d+(a.hash||"")};function pb(a){a:{try{var b=a.document.cookie}catch(h){b=""}if(a=b)for(a=a.split(";"),b=0;b<a.length;b++){var c=a[b].trim(),d=c.indexOf("="),e;if(e=-1!=d)e=c.substring(0,d).trim(),e="AMP_EXP"==z(e,void 0);if(e){a=c.substring(d+1).trim();a=z(a,a);break a}}a=null}var f=a,g=f?f.split(/\s*,\s*/g):[];a=Object.create(null);for(b=0;b<g.length;b++)0!=g[b].length&&("-"==g[b][0]?a[g[b].substr(1)]=!1:a[g[b]]=!0);return a};function qb(a){return!!a&&"function"==typeof a.getFormData};var rb=["GET","POST"],sb=[B,function(a){return"[object Object]"===ya.call(a)}];function tb(a){var b="application/json";a=a||{};var c=a.method;void 0===c?c="GET":(c=c.toUpperCase(),rb.includes(c));a.method=c;a.headers=a.headers||G({});b&&(a.headers.Accept=b);return a}
-function ub(a){var b=tb(a);if("POST"==b.method&&!qb(b.body)){sb.some(function(a){return a(b.body)});b.headers["Content-Type"]=b.headers["Content-Type"]||"text/plain;charset=utf-8";var c=b.headers["Content-Type"];b.body="application/x-www-form-urlencoded"===c?mb(b.body):JSON.stringify(b.body)}}
-function vb(a){var b=a.getAttribute("crossorigin");return b&&"amp-viewer-auth-token-via-post"===b.trim()?Za(a,"amp-viewer-assistance","amp-viewer-assistance").then(function(a){F(a,'crossorigin="amp-viewer-auth-token-post" requires amp-viewer-assistance extension.');return a.getIdTokenPromise()}).then(function(a){return a||""}).catch(function(){return""}):Promise.resolve(void 0)};function T(a,b){var c="amp-list";this.U=a;this.m=b;this.ea=c}T.prototype.isSupported=function(){var a=this.U.ampdoc;return a.isSingleDoc()&&a.getRootNode().documentElement.hasAttribute("allow-viewer-render-template")?this.U.hasCapability("viewerRenderTemplate"):!1};T.prototype.fetchAndRenderTemplate=function(a,b,c,d){c=void 0===c?null:c;d=void 0===d?{}:d;var e;c||(e=this.m.maybeFindTemplate(a));return this.U.sendMessageAwaitResponse("viewerRenderTemplate",wb(this,b,e,c,d))};
-T.prototype.renderTemplate=function(a,b){if(this.isSupported()){F("string"===typeof b.html,"Server side html response must be defined");var c=this.m.findAndSetHtmlForTemplate(a,b.html)}else c=B(b)?this.m.findAndRenderTemplateArray(a,b):this.m.findAndRenderTemplate(a,b);return c};
-function wb(a,b,c,d,e){e=void 0===e?{}:e;var f=G({type:a.ea}),g="successTemplate",h=d&&d[g]?d[g]:c;h&&(f[g]={type:"amp-mustache",payload:h.innerHTML});var k="errorTemplate",l=d&&d[k]?d[k]:null;l&&(f[k]={type:"amp-mustache",payload:l.innerHTML});e&&Object.assign(f,e);a=b.xhrUrl;c=b.fetchOpt;b=Object.assign({},c);if(qb(c.body)){c=c.body;b.headers["Content-Type"]="multipart/form-data;charset=utf-8";c=c.entries();d=[];for(e=c.next();!e.done;e=c.next())d.push(e.value);b.body=d}return G({originalRequest:{input:a,
-init:b},ampComponent:f})};function xb(a,b){if("."==b)return a;b=b.split(".");for(var c=0;c<b.length;c++){var d=b[c];if(d&&a&&void 0!==a[d]&&(null==a||"object"!=typeof a?0:Object.prototype.hasOwnProperty.call(a,d)))a=a[d];else{a=void 0;break}}return a};function yb(a,b,c,d,e){var f=".";f=void 0===f?".":f;c=void 0===c?0:c;d=void 0===d?!1:d;nb(b.getAttribute("src"),b);var g=O(a.win,"batched-xhr");return zb(b,c,d).then(function(a){void 0!==e&&(a.fetchOpt.method="POST",a.fetchOpt.headers={"Content-Type":"application/x-www-form-urlencoded"},a.fetchOpt.body={ampViewerAuthToken:e});return g.fetchJson(a.xhrUrl,a.fetchOpt)}).then(function(a){return a.json()}).then(function(a){if(null==a)throw Error("Response is undefined.");return xb(a,f||".")})}
-function zb(a,b,c){var d=a.getAttribute("src"),e=Ra(a,"url-replace");return(1<=b?e.expandUrlAsync(d):Promise.resolve(d)).then(function(d){if(1==b){var f=e.collectUnwhitelistedVarsSync(a);if(0<f.length)throw E().createError("URL variable substitutions in CORS fetches from dynamic URLs (e.g. via amp-bind) require opt-in. "+('Please add data-amp-replace="'+f.join(" ")+'" to the ')+("<"+a.tagName+"> element. See https://bit.ly/amp-var-subs."));}var h={};a.hasAttribute("credentials")&&(h.credentials=a.getAttribute("credentials"));
-c&&(h.cache="reload");return{xhrUrl:d,fetchOpt:h}})};var U;function Ab(a,b){var c=void 0,d=a,e=b;var f=function(a){try{return e(a)}catch(l){throw self.reportError(l),l;}};var g=Bb(),h=!1;c&&(h=c.capture);d.addEventListener("click",f,g?c:h)}function Bb(){if(void 0!==U)return U;U=!1;try{var a={get capture(){U=!0}};self.addEventListener("test-options",null,a);self.removeEventListener("test-options",null,a)}catch(b){}return U};function Cb(a,b,c,d){var e={detail:c};Object.assign(e,d);if("function"==typeof a.CustomEvent)return new a.CustomEvent(b,e);a=a.document.createEvent("CustomEvent");a.initCustomEvent(b,!!e.bubbles,!!e.cancelable,c);return a}function Db(a,b){Ab(a,b)};function V(a){var b=AMP.BaseElement.call(this,a)||this;b.h=null;b.F=null;b.V=!1;b.X=new P(b.win,function(){return Eb(b)});b.A=null;b.Y=null;b.m=O(b.win,"templates");b.L=!1;b.W=null;b.B=null;b.o=!1;b.O=null;b.w=null;b.R=!1;b.$=null;b.registerAction("refresh",function(){if(b.L)return Fb(b),W(b,!1,!0)},100);b.registerAction("changeToLayoutContainer",function(){return Gb(b)},100);b.I=null;return b}var X=AMP.BaseElement;V.prototype=aa(X.prototype);V.prototype.constructor=V;
-if(da)da(V,X);else for(var Y in X)if("prototype"!=Y)if(Object.defineProperties){var Hb=Object.getOwnPropertyDescriptor(X,Y);Hb&&Object.defineProperty(V,Y,Hb)}else V[Y]=X[Y];V.fa=X.prototype;m=V.prototype;m.isLayoutSupported=function(a){return"fixed"==a||"fixed-height"==a||"responsive"==a||"fill"==a||"flex-item"==a||"fluid"==a||"intrinsic"==a};
-m.buildCallback=function(){var a=this;this.F=this.getViewport();var b=Ta(this.getAmpDoc());this.I=new T(b,this.m);this.o=this.element.hasAttribute("load-more");this.W=this.element.getAttribute("src");this.h=Ib(this);this.element.appendChild(this.h);this.element.hasAttribute("aria-live")||this.element.setAttribute("aria-live","polite");this.element.hasAttribute("auto-resize")&&E().warn("amp-list","auto-resize attribute is deprecated and its behavior is disabled. This feature will be relaunched under a new name soon. Please see https://github.com/ampproject/amphtml/issues/18849");
-ab(this.element).then(function(b){a.B=b})};m.reconstructWhenReparented=function(){return!1};m.layoutCallback=function(){var a=this;this.L=!0;var b=this.getPlaceholder();b&&Jb(this,b);this.F.onResize(function(){Kb(a)});this.o&&Lb(this);return W(this)};
-function Lb(a){a.mutateElement(function(){Z(a).initializeLoadMore();var b=a.getOverflowElement();b&&Da(b);a.element.warnOnMissingOverflow=!1}).then(function(){Mb(a);Db(Z(a).getLoadMoreFailedClickable(),function(){return Nb(a,!0)});Db(Z(a).getLoadMoreButtonClickable(),function(){return Nb(a)})})}function Kb(a){a.o?Ob(a,a.h):Jb(a,a.h)}function Z(a){a.O||(a.O=new Na(a.element));return a.O}
-function Mb(a){var b,c;a.measureMutateElement(function(){b=Z(a).getLoadMoreButton().offsetHeight;c=a.element.offsetHeight},function(){J(a.h,{"max-height":"calc(100% - "+(b+"px")+")"});a.element.changeSize(c+b)})}
-m.mutatedAttributesCallback=function(a){function b(a){c.element.setAttribute("src","");a=B(a)?a:[a];Fb(c,!1);return Pb(c,a,!1)}var c=this;za().info("amp-list","mutate:",this.element,a);var d=a.src,e=a.state;if(void 0!==d)if("string"===typeof d){if(this.L){Fb(this);var f=W(this)}}else"object"===typeof d?f=b(d):this.user().error("amp-list",'Unexpected "src" type: '+d);else void 0!==e&&(E().error("amp-list","[state] is deprecated, please use [src] instead."),f=b(e));var g=a["is-layout-container"];g&&
-Gb(this);if(xa().test)return f};m.isLoadingReused=function(){return this.element.hasAttribute("reset-on-refresh")};function Ib(a){var b=a.win.document.createElement("div");b.setAttribute("role","list");a.o||a.applyFillContent(b,!0);return b}function Qb(a,b){a.forEach(function(a){a.hasAttribute("role")||a.setAttribute("role","listitem");b.appendChild(a)})}function Rb(a,b){if(b||a.V)a.toggleFallback(b),a.V=b}
-function Fb(a,b){if((b=void 0===b?!0:b)&&a.element.hasAttribute("reset-on-refresh")||"always"===a.element.getAttribute("reset-on-refresh"))a.togglePlaceholder(!0),a.toggleLoading(!0,!0),a.mutateElement(function(){Rb(a,!1);if(a.B){var b=(b=a.h.children)?Array.prototype.slice.call(b):[];a.B.rescan([],b,{fast:!0,update:!1})}Ja(a.h)})}
-function W(a,b,c){b=void 0===b?!1:b;c=void 0===c?!1:c;if(!a.element.getAttribute("src"))return Promise.resolve();if(a.I.isSupported())c=Sb(a,c);else{var d=a.element.getAttribute("items")||"items";c=Tb(a,c).then(function(c){var e=c;"."!=d&&(e=xb(c,d));F("undefined"!==typeof e,'Response must contain an array or object at "%s". %s',d,a.element);a.element.hasAttribute("single-item")&&!B(e)&&(e=[e]);e=E().assertArray(e);if(a.element.hasAttribute("max-items")){var g=e,h=parseInt(a.element.getAttribute("max-items"),
-10);h<g.length&&(g=g.slice(0,h));e=g}a.o&&(g=a.element.getAttribute("load-more-bookmark")||"load-more-src",a.w=xb(c,g));return Pb(a,e,b,c)})}return c.catch(function(c){var d=c?Cb(a.win,"amp-list.error",G({response:c.response})):null,e=Ra(a.element,"action");e.trigger(a.element,"fetch-error",d,1);if(b)throw c;a.toggleLoading(!1);if(a.getFallback())Rb(a,!0),a.togglePlaceholder(!1);else throw c;})}
-function Sb(a,b){var c;return zb(a.element,Ub(a),b).then(function(b){var d=c=b,f=c.xhrUrl;if(!1!==c.fetchOpt.ampCors){var g=a.win,h=S(f);h=A(h.search);F(!("__amp_source_origin"in h),"Source origin is not allowed in %s",f);g=S(ob(g.location.href)).origin;g=encodeURIComponent("__amp_source_origin")+"="+encodeURIComponent(g);g&&(f=f.split("#",2),h=f[0].split("?",2),g=h[0]+(h[1]?"?"+h[1]+"&"+g:"?"+g),f=g+=f[1]?"#"+f[1]:"")}d.xhrUrl=f;d=c;g=c.xhrUrl;f=(f=c.fetchOpt)||{};h=a.win;h=h.origin||S(h.location.href).origin;
-g=S(g).origin;h==g&&(f.headers=f.headers||{},f.headers["AMP-Same-Origin"]="true");d.fetchOpt=f;ub(b.fetchOpt);b=G({ampListAttributes:{items:a.element.getAttribute("items")||"items",singleItem:a.element.hasAttribute("single-item"),maxItems:a.element.getAttribute("max-items")}});return a.I.fetchAndRenderTemplate(a.element,c,null,b)}).then(function(a){F(a&&"string"===typeof a.html,"Expected response with format {html: <string>}. Received: ",a);c.fetchOpt.responseType="application/json";return a},function(a){throw E().createError("Error proxying amp-list templates",
-a);}).then(function(b){return Pb(a,b,!1)})}function Pb(a,b,c,d){var e=new ua,f=e.promise,g=e.resolve,h=e.reject;a.A||a.X.schedule();a.A={data:b,resolver:g,rejecter:h,append:c,payload:d};a.Y&&c&&(a.A.payload=d||{});return f}
-function Eb(a){function b(){d();e.rejecter()}function c(){d();e.resolver()}function d(){a.A!==e?a.X.schedule(1):(a.Y=a.A.data,a.A=null)}var e=a.A,f=a.I.isSupported(),g=a.I.renderTemplate(a.element,e.data).then(function(b){return Vb(a,b,e.append)}).then(function(b){return Wb(a,b,e.append)});if(!f){var h=e.payload;g=g.then(function(){if(a.o){var b=[];b.push(Xb(a,Z(a).getLoadMoreButton(),h));b.push(Xb(a,Z(a).getLoadMoreEndElement(),h));b=Promise.all(b)}else b=Promise.resolve();return b}).then(function(){return a.o?
-Yb(a):Promise.resolve()})}g.then(c,b)}function Xb(a,b,c){return b&&a.m.hasTemplate(b)?a.m.findAndRenderTemplate(b,c).then(function(c){return a.mutateElement(function(){Ja(b);b.appendChild(c)})}):Promise.resolve()}
-function Vb(a,b,c){function d(b){var d=c?[]:[a.h];return b.rescan(e,d,{fast:!0,update:!0}).then(function(){return e},function(){return e})}var e=B(b)?b:[b],f=a.element.getAttribute("binding");if("no"===f)return Promise.resolve(e);var g=e.some(function(a){return a.hasAttribute("i-amphtml-binding")||!!a.querySelector("[i-amphtml-binding]")});if(!g)return Promise.resolve(e);if("refresh"===f){if(a.B&&a.B.signals().get("FIRST_MUTATE"))return d(a.B);ab(a.element).then(function(a){a&&a.rescan(e,[],{fast:!0,
-update:!1})});return Promise.resolve(e)}return ab(a.element).then(function(a){return a?d(a):Promise.resolve(e)})}
-function Wb(a,b,c){c=void 0===c?!1:c;za().info("amp-list","render:",a.element,b);var d=a.h;return a.mutateElement(function(){a.toggleLoading(!1);a.getFallback()&&Rb(a,!1);a.togglePlaceholder(!1);var e=a.win;if(e.__AMP__EXPERIMENT_TOGGLES)var f=e.__AMP__EXPERIMENT_TOGGLES;else{e.__AMP__EXPERIMENT_TOGGLES=Object.create(null);f=e.__AMP__EXPERIMENT_TOGGLES;if(e.AMP_CONFIG)for(var g in e.AMP_CONFIG){var h=e.AMP_CONFIG[g];"number"===typeof h&&0<=h&&1>=h&&(f[g]=Math.random()<h)}if(e.AMP_CONFIG&&Array.isArray(e.AMP_CONFIG["allow-doc-opt-in"])&&
-0<e.AMP_CONFIG["allow-doc-opt-in"].length&&(g=e.AMP_CONFIG["allow-doc-opt-in"],h=e.document.head.querySelector('meta[name="amp-experiments-opt-in"]'))){h=h.getAttribute("content").split(",");for(var k=0;k<h.length;k++)-1!=g.indexOf(h[k])&&(f[h[k]]=!0)}Object.assign(f,pb(e));if(e.AMP_CONFIG&&Array.isArray(e.AMP_CONFIG["allow-url-opt-in"])&&0<e.AMP_CONFIG["allow-url-opt-in"].length)for(g=e.AMP_CONFIG["allow-url-opt-in"],e=A(e.location.originalHash||e.location.hash),h=0;h<g.length;h++)k=e["e-"+g[h]],
-"1"==k&&(f[g[h]]=!0),"0"==k&&(f[g[h]]=!1)}var l=!!f["amp-list-diffing"];if(l&&d.hasChildNodes()){var D=Ib(a);Qb(b,D);var gb=w.default||w;gb.KEY="i-amphtml-key";gb(d,D)}else c||Ja(d),Qb(b,d);f=Cb(a.win,"amp:dom-update",null,{bubbles:!0});a.h.dispatchEvent(f);a.element.getResources().getResourceForElement(a.element).resetPendingChangeSize();Kb(a)})}
-function Jb(a,b){"container"!=a.element.getAttribute("layout")&&a.measureElement(function(){var c=b.scrollHeight;c>a.element.offsetHeight&&a.attemptChangeHeight(c).catch(function(){})})}function Ob(a,b){var c=a.w?Z(a).getLoadMoreButton():Z(a).getLoadMoreEndElement();Zb(a,c,b)}
-function Zb(a,b,c){"container"!=a.element.getAttribute("layout")&&a.measureElement(function(){var d=c.scrollHeight,e=b?b.offsetHeight:0;d+e>a.element.offsetHeight&&a.attemptChangeHeight(d+e).then(function(){a.R=!1;"auto"===a.element.getAttribute("load-more")&&$b(a);J(a.h,{"max-height":""})}).catch(function(){a.R=!0;Mb(a)})})}
-function Gb(a){var b=a.element.getAttribute("layout");return"container"==b?Promise.resolve():a.mutateElement(function(){switch(b){case "responsive":a.element.classList.remove("i-amphtml-layout-responsive");J(a.element,{height:"",width:""});break;case "flex-item":J(a.element,{height:"",width:""});break;case "fixed":a.element.classList.remove("i-amphtml-layout-fixed");J(a.element,{height:""});break;case "fixed-height":a.element.classList.remove("i-amphtml-layout-fixed-height");J(a.element,{height:"",
-width:""});break;case "intrinsic":a.element.classList.remove("i-amphtml-layout-intrinsic")}a.element.changeSize();a.element.classList.remove("i-amphtml-layout-size-defined");a.h.classList.remove("i-amphtml-fill-content","i-amphtml-replaced-content");var c=a.getOverflowElement();c&&Da(c);a.element.classList.add("i-amphtml-layout-container");a.element.setAttribute("layout","container")})}
-function Yb(a){if(a.w){var b="auto"===a.element.getAttribute("load-more");b&&ac(a);return a.mutateElement(function(){Z(a).toggleLoadMoreLoading(!1);J(Z(a).getLoadMoreButton(),{visibility:""})})}return a.mutateElement(function(){return Z(a).setLoadMoreEnded()})}
-function Nb(a,b){b=void 0===b?!1:b;if(a.w)a.element.setAttribute("src",a.w),a.w=null;else if(!b)return Promise.resolve();a.mutateElement(function(){Z(a).toggleLoadMoreLoading(!0)});return W(a,!0).then(function(){return a.mutateElement(function(){a.w?Z(a).toggleLoadMoreLoading(!1):Z(a).setLoadMoreEnded()})}).then(function(){Ob(a,a.h)}).catch(function(){bc(a)})}function bc(a){a.mutateElement(function(){return Z(a).setLoadMoreFailed()}).then(function(){Zb(a,Z(a).getLoadMoreFailedElement(),a.h)})}
-function ac(a){a.$||(a.$=a.F.onChanged(function(){return $b(a)}))}function $b(a){if(!a.R){var b=a.h.lastChild||a.h;a.F.getClientRectAsync(b).then(function(b){var c=a.F.getHeight(),e=a.F.getScrollTop();if(e+3*c>b.bottom)return Nb(a)})}}function Tb(a,b){b=void 0===b?!1:b;return vb(a.element).then(function(c){var d=b;d=void 0===d?!1:d;return yb(a.getAmpDoc(),a.element,Ub(a),d,c)})}
-function Ub(a){var b=a.element.getAttribute("src"),c=1;if(b==a.W||S(ob(b)).origin==S(ob(a.getAmpDoc().win.location)).origin)c=2;return c}(function(a){a.registerElement("amp-list",V,"amp-list[load-more] [load-more-button].amp-visible,amp-list[load-more] [load-more-end].amp-visible,amp-list[load-more] [load-more-failed].amp-visible,amp-list[load-more] [load-more-loading].amp-visible{display:block}amp-list[load-more] [load-more-button].i-amphtml-default-ui,amp-list[load-more] [load-more-failed].i-amphtml-default-ui,amp-list[load-more] [load-more-loading].i-amphtml-default-ui{height:80px;padding:12px 0px;box-sizing:border-box}.i-amphtml-list-load-more-button,amp-list[load-more] [load-more-button].i-amphtml-default-ui,amp-list[load-more] [load-more-failed].i-amphtml-default-ui,amp-list[load-more] [load-more-loading].i-amphtml-default-ui{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-weight:700;font-size:14px;text-transform:uppercase;letter-spacing:.1em;color:#333;text-align:center}amp-list[load-more] [load-more-loading].i-amphtml-default-ui .i-amphtml-list-load-more-spinner{display:inline-block;width:40px;height:40px;margin:8px 0px;background-image:url(\"data:image/svg+xml;charset=utf-8,%3Csvg viewBox='0 0 40 40' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3ClinearGradient id='a'%3E%3Cstop stop-color='%23333' stop-opacity='.75'/%3E%3Cstop offset='100%25' stop-color='%23333' stop-opacity='0'/%3E%3C/linearGradient%3E%3C/defs%3E%3Cpath d='M11 4.4A18 18 0 1 0 38 20' fill='none' stroke='url(%23a)' stroke-width='1.725'/%3E%3C/svg%3E\");animation:amp-list-load-more-spinner 1000ms linear infinite}@keyframes amp-list-load-more-spinner{0%{transform:rotate(0deg)}to{transform:rotate(360deg)}}.i-amphtml-list-load-more-button{border:none;display:inline-block;background-color:rgba(51,51,51,0.75);color:#fff;margin:4px 0px;padding:0px 32px;box-sizing:border-box;height:48px;border-radius:24px}[load-more] div[role=list]{overflow-y:hidden}.i-amphtml-list-load-more-button,.i-amphtml-list-load-more-button label,.i-amphtml-list-load-more-icon{cursor:pointer}.i-amphtml-list-load-more-button:hover{background-color:#333}.i-amphtml-list-load-more-button.i-amphtml-list-load-more-button-small{margin:0px;padding:4px 16px;height:32px}.i-amphtml-list-load-more-button label{display:inline-block;vertical-align:middle;line-height:24px}amp-list[load-more] [load-more-failed].i-amphtml-default-ui .i-amphtml-list-load-more-message{line-height:24px}amp-list[load-more] [load-more-failed].i-amphtml-default-ui .i-amphtml-list-load-more-icon{height:24px;width:24px;display:inline-block;vertical-align:middle;background-image:url(\"data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24'%3E%3Cpath fill='%23fff' d='M17.65 6.35A7.958 7.958 0 0 0 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08A5.99 5.99 0 0 1 12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z'/%3E%3C/svg%3E\")}\n/*# sourceURL=/extensions/amp-list/0.1/amp-list.css*/")})(self.AMP);
+var na;function oa(a,b,c){if(b(a))c();else{var d=a.ownerDocument.defaultView;if(d.MutationObserver){var e=new d.MutationObserver(function(){b(a)&&(e.disconnect(),c())});e.observe(a,{childList:!0})}else var g=d.setInterval(function(){b(a)&&(d.clearInterval(g),c())},5)}}function pa(a,b){oa(a.documentElement,function(){return!!a.body},b)}function qa(a){return new Promise(function(b){return pa(a,b)})}function ra(a){for(;a.firstChild;)a.removeChild(a.firstChild)}
+function I(a,b){/^[\w-]+$/.test(b);b="> ["+b+"]";if(void 0!==na)var c=na;else{try{var d=a.ownerDocument,e=d.createElement("div"),g=d.createElement("div");e.appendChild(g);c=e.querySelector(":scope div")===g}catch(h){c=!1}c=na=c}c?a=a.querySelector(b.replace(/^|,/g,"$&:scope ")):(a.classList.add("i-amphtml-scoped"),b=b.replace(/^|,/g,"$&.i-amphtml-scoped "),b=a.querySelectorAll(b),a.classList.remove("i-amphtml-scoped"),a=void 0===b[0]?null:b[0]);return a};var sa=['\n        <amp-list-load-more load-more-failed class="i-amphtml-default-ui">\n          <div class="i-amphtml-list-load-more-message">\n            Unable to Load More\n          </div>\n          <button load-more-clickable\n            class="i-amphtml-list-load-more-button\n                  i-amphtml-list-load-more-button-has-icon\n                  i-amphtml-list-load-more-button-small"\n          >\n            <div class="i-amphtml-list-load-more-icon"></div>\n            <label>Retry</label>\n          </button>\n        </amp-list-load-more>\n      '];
+sa.raw=sa.slice();var ta=['\n        <amp-list-load-more load-more-loading class="i-amphtml-default-ui">\n          <div class="i-amphtml-list-load-more-spinner"></div>\n        </amp-list-load-more>\n      '];ta.raw=ta.slice();var ua=['\n        <amp-list-load-more load-more-button\n          class="amp-visible i-amphtml-default-ui">\n          <button load-more-clickable class="i-amphtml-list-load-more-button">\n            <label>See More</label>\n          </button>\n        </amp-list-load-more>\n      '];
+ua.raw=ua.slice();function va(a){this.j=a;this.G=this.N=this.C=this.D=this.M=this.l=null}n=va.prototype;n.initializeLoadMore=function(){wa(this);xa(this);ya(this);this.G||(this.G=I(this.j,"load-more-end"))&&this.j.appendChild(this.G)};function wa(a){a.l=I(a.j,"load-more-button");a.l?a.l.classList.add("amp-visible"):a.l=F(a.j)(ua);a.j.appendChild(a.l);H(a.l,{visibility:"hidden"})}function xa(a){a.D=I(a.j,"load-more-loading");a.D||(a.D=F(a.j)(ta));a.j.appendChild(a.D)}
+n.getLoadMoreButton=function(){this.l||wa(this);return this.l};n.getLoadMoreLoadingElement=function(){this.D||xa(this);return this.D};n.getLoadMoreButtonClickable=function(){if(!this.M){var a=this.getLoadMoreButton();this.M=I(a,"load-more-clickable")||a}return this.M};function ya(a){a.C=I(a.j,"load-more-failed");a.C||(a.C=F(a.j)(sa));a.j.appendChild(a.C)}n.getLoadMoreFailedElement=function(){this.C||ya(this);return this.C};
+n.getLoadMoreFailedClickable=function(){if(!this.N){var a=this.getLoadMoreFailedElement();this.N=I(a,"load-more-clickable")||a}return this.N};n.getLoadMoreEndElement=function(){return this.G};n.setLoadMoreEnded=function(){this.getLoadMoreFailedElement().classList.toggle("amp-visible",!1);this.getLoadMoreButton().classList.toggle("amp-visible",!1);this.getLoadMoreLoadingElement().classList.toggle("amp-visible",!1);var a=this.getLoadMoreEndElement();a&&a.classList.toggle("amp-visible",!0)};
+n.toggleLoadMoreLoading=function(a){a&&this.getLoadMoreFailedElement().classList.toggle("amp-visible",!1);var b=this.getLoadMoreEndElement();b&&b.classList.toggle("amp-visible",!1);this.getLoadMoreButton().classList.toggle("amp-visible",!a);this.getLoadMoreLoadingElement().classList.toggle("amp-visible",a)};
+n.setLoadMoreFailed=function(){var a=this.getLoadMoreFailedElement(),b=this.getLoadMoreButton();if(a||b)a.classList.toggle("amp-visible",!0),b.classList.toggle("amp-visible",!1),this.getLoadMoreLoadingElement().classList.toggle("amp-visible",!1)};function za(a,b){var c=a.ownerDocument.defaultView,d=c!=(c.__AMP_TOP||c);d?b=Aa(c,b)?J(c,b):null:(a=K(a),a=L(a),b=Aa(a,b)?J(a,b):null);return b}function M(a,b){a=a.__AMP_TOP||a;return J(a,b)}function Ba(a){var b=K(a);b=L(b);return J(b,"viewer")}function Ca(a,b){return Da(L(a),b)}function K(a){return a.nodeType?M((a.ownerDocument||a).defaultView,"ampdoc").getAmpDoc(a):a}function L(a){a=K(a);return a.isSingleDoc()?a.win:a}
+function J(a,b){Aa(a,b);var c=Ea(a);a=c[b];a.obj||(a.obj=new a.ctor(a.context),a.ctor=null,a.context=null,a.resolve&&a.resolve(a.obj));return a.obj}function Fa(a,b){var c=Da(a,b);if(c)return c;a=Ea(a);a[b]=Ga();return a[b].promise}function Da(a,b){var c=Ea(a)[b];if(c){if(c.promise)return c.promise;J(a,b);return c.promise=Promise.resolve(c.obj)}return null}function Ea(a){var b=a.services;b||(b=a.services={});return b}function Aa(a,b){a=a.services&&a.services[b];return!(!a||!a.ctor&&!a.obj)}
+function Ga(){var a=new ea,b=a.promise,c=a.resolve;a=a.reject;b.catch(function(){});return{obj:null,promise:b,resolve:c,reject:a,context:null,ctor:null}};function Ha(a,b,c){var d=Ca(a,b);if(d)return d;var e=K(a);return e.waitForBodyOpen().then(function(){return Ia(e.win,c,e.win.document.head)}).then(function(){var d=e.win;return d.ampExtendedElements&&d.ampExtendedElements[c]?Fa(L(a),b):null})}function Ja(a){var b=za(a,"bind");if(b)return Promise.resolve(b);b=a.ownerDocument.defaultView;return b!==(b.__AMP_TOP||b)?Ka(b):Ha(a,"bind","amp-bind")}
+function Ia(a,b,c){if(c){var d=[];c=c.querySelectorAll("script[custom-element]");for(var e=0;e<c.length;e++)d.push(c[e].getAttribute("custom-element"))}else d=[];if(!d.includes(b))return Promise.resolve();var g=M(a,"extensions");return g.waitForExtension(a,b)}function Ka(a){return qa(a.document).then(function(){return Ia(a,"amp-bind",a.document.head)}).then(function(){return a.ampExtendedElements&&a.ampExtendedElements["amp-bind"]?Fa(a,"bind"):null})};function N(a,b){var c,d=this;this.Z=M(a,"timer");this.da=b;this.ca=c||0;this.H=-1;this.P=0;this.S=!1;this.aa=function(){d.H=-1;d.P=0;d.S=!0;d.da();d.S=!1}}N.prototype.isPending=function(){return-1!=this.H};N.prototype.schedule=function(a){var b=a||this.ca;this.S&&10>b&&(b=10);var c=Date.now()+b;return!this.isPending()||-10>c-this.P?(this.cancel(),this.P=c,this.H=this.Z.delay(this.aa,b),!0):!1};N.prototype.cancel=function(){this.isPending()&&(this.Z.cancel(this.H),this.H=-1)};function O(){var a=100;this.ba=a;this.K=this.T=0;this.J=Object.create(null)}O.prototype.has=function(a){return!!this.J[a]};O.prototype.get=function(a){var b=this.J[a];if(b)return b.access=++this.K,b.payload};O.prototype.put=function(a,b){this.has(a)||this.T++;this.J[a]={payload:b,access:this.K};if(!(this.T<=this.ba)){B().warn("lru-cache","Trimming LRU cache");a=this.J;var c=this.K+1,d;for(d in a){var e=a[d].access;if(e<c){c=e;var g=d}}void 0!==g&&(delete a[g],this.T--)}};var P=self.AMP_CONFIG||{},La={thirdParty:P.thirdPartyUrl||"https://3p.ampproject.net",thirdPartyFrameHost:P.thirdPartyFrameHost||"ampproject.net",thirdPartyFrameRegex:("string"==typeof P.thirdPartyFrameRegex?new RegExp(P.thirdPartyFrameRegex):P.thirdPartyFrameRegex)||/^d-\d+\.ampproject\.net$/,cdn:P.cdnUrl||"https://cdn.ampproject.org",cdnProxyRegex:("string"==typeof P.cdnProxyRegex?new RegExp(P.cdnProxyRegex):P.cdnProxyRegex)||/^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org$/,localhostRegex:/^https?:\/\/localhost(:\d+)?$/,
+errorReporting:P.errorReportingUrl||"https://amp-error-reporting.appspot.com/r",localDev:P.localDev||!1,trustedViewerHosts:[/(^|\.)google\.(com?|[a-z]{2}|com?\.[a-z]{2}|cat)$/,/(^|\.)gmail\.dev$/]};var Ma=D({c:!0,v:!0,a:!0,ad:!0,action:!0}),Na,Oa,Pa=/[?&]amp_js[^&]*/,Qa=/[?&]amp_gsa[^&]*/,Ra=/[?&]amp_r[^&]*/,Sa=/[?&]amp_kit[^&]*/,Ta=/[?&]usqp[^&]*/;
+function Q(a){var b;Na||(Na=self.document.createElement("a"),Oa=self.UrlCache||(self.UrlCache=new O));var c=b?null:Oa,d=Na;if(c&&c.has(a))a=c.get(a);else{d.href=a;d.protocol||(d.href=d.href);var e={href:d.href,protocol:d.protocol,host:d.host,hostname:d.hostname,port:"0"==d.port?"":d.port,pathname:d.pathname,search:d.search,hash:d.hash,origin:null};"/"!==e.pathname[0]&&(e.pathname="/"+e.pathname);if("http:"==e.protocol&&80==e.port||"https:"==e.protocol&&443==e.port)e.port="",e.host=e.hostname;e.origin=
+d.origin&&"null"!=d.origin?d.origin:"data:"!=e.protocol&&e.host?e.protocol+"//"+e.host:e.href;d=ha().test&&Object.freeze?Object.freeze(e):e;c&&c.put(a,d);a=d}return a}function Ua(a){var b=[],c;for(c in a){var d=a[c];if(null!=d)if(x(d))for(var e=0;e<d.length;e++){var g=d[e];b.push(encodeURIComponent(c)+"="+encodeURIComponent(g))}else e=d,b.push(encodeURIComponent(c)+"="+encodeURIComponent(e))}return b.join("&")}
+function Va(a,b){var c=void 0===c?"source":c;C(null!=a,"%s %s must be available",b,c);var d=a;a=d;"string"==typeof a&&(a=Q(a));var e;(e="https:"==a.protocol||"localhost"==a.hostname||"127.0.0.1"==a.hostname)||(a=a.hostname,e=a.length-10,e=0<=e&&a.indexOf(".localhost",e)==e);C(e||/^(\/\/)/.test(d),'%s %s must start with "https://" or "//" or be relative and served from either https or from localhost. Invalid value: %s',b,c,d)}
+function Wa(a){"string"==typeof a&&(a=Q(a));var b=a;"string"==typeof b&&(b=Q(b));if(!La.cdnProxyRegex.test(b.origin))return a.href;b=a.pathname.split("/");C(Ma[b[1]],"Unknown path prefix in url %s",a.href);var c=b[2],d="s"==c?"https://"+decodeURIComponent(b[3]):"http://"+decodeURIComponent(c);C(0<d.indexOf("."),"Expected a . in origin %s",d);b.splice(1,"s"==c?3:2);b=d+b.join("/");d=(d=a.search)&&"?"!=d?(d=d.replace(Pa,"").replace(Qa,"").replace(Ra,"").replace(Sa,"").replace(Ta,"").replace(/^[?&]/,
+""))?"?"+d:"":"";return b+d+(a.hash||"")};function Xa(a){a:{try{var b=a.document.cookie}catch(f){b=""}if(a=b)for(a=a.split(";"),b=0;b<a.length;b++){var c=a[b].trim(),d=c.indexOf("="),e;if(e=-1!=d)e=c.substring(0,d).trim(),e="AMP_EXP"==u(e,void 0);if(e){a=c.substring(d+1).trim();a=u(a,a);break a}}a=null}var g=a,h=g?g.split(/\s*,\s*/g):[];a=Object.create(null);for(b=0;b<h.length;b++)0!=h[b].length&&("-"==h[b][0]?a[h[b].substr(1)]=!1:a[h[b]]=!0);return a};function Ya(a){return!!a&&"function"==typeof a.getFormData};var Za=["GET","POST"],$a=[x,function(a){return"[object Object]"===ia.call(a)}];function ab(a){var b="application/json";a=a||{};var c=a.method;void 0===c?c="GET":(c=c.toUpperCase(),Za.includes(c));a.method=c;a.headers=a.headers||D({});b&&(a.headers.Accept=b);return a}
+function bb(a){var b=ab(a);if("POST"==b.method&&!Ya(b.body)){$a.some(function(a){return a(b.body)});b.headers["Content-Type"]=b.headers["Content-Type"]||"text/plain;charset=utf-8";var c=b.headers["Content-Type"];b.body="application/x-www-form-urlencoded"===c?Ua(b.body):JSON.stringify(b.body)}}
+function cb(a){var b=a.getAttribute("crossorigin");return b&&"amp-viewer-auth-token-via-post"===b.trim()?Ha(a,"amp-viewer-assistance","amp-viewer-assistance").then(function(a){C(a,'crossorigin="amp-viewer-auth-token-post" requires amp-viewer-assistance extension.');return a.getIdTokenPromise()}).then(function(a){return a||""}).catch(function(){return""}):Promise.resolve(void 0)};function R(a,b){var c="amp-list";this.U=a;this.m=b;this.ea=c}R.prototype.isSupported=function(){var a=this.U.ampdoc;return a.isSingleDoc()&&a.getRootNode().documentElement.hasAttribute("allow-viewer-render-template")?this.U.hasCapability("viewerRenderTemplate"):!1};R.prototype.fetchAndRenderTemplate=function(a,b,c,d){c=void 0===c?null:c;d=void 0===d?{}:d;var e;c||(e=this.m.maybeFindTemplate(a));return this.U.sendMessageAwaitResponse("viewerRenderTemplate",db(this,b,e,c,d))};
+R.prototype.renderTemplate=function(a,b){if(this.isSupported()){C("string"===typeof b.html,"Server side html response must be defined");var c=this.m.findAndSetHtmlForTemplate(a,b.html)}else c=x(b)?this.m.findAndRenderTemplateArray(a,b):this.m.findAndRenderTemplate(a,b);return c};
+function db(a,b,c,d,e){e=void 0===e?{}:e;var g=D({type:a.ea}),h="successTemplate",f=d&&d[h]?d[h]:c;f&&(g[h]={type:"amp-mustache",payload:f.innerHTML});var k="errorTemplate",l=d&&d[k]?d[k]:null;l&&(g[k]={type:"amp-mustache",payload:l.innerHTML});e&&Object.assign(g,e);a=b.xhrUrl;c=b.fetchOpt;b=Object.assign({},c);if(Ya(c.body)){c=c.body;b.headers["Content-Type"]="multipart/form-data;charset=utf-8";c=c.entries();d=[];for(e=c.next();!e.done;e=c.next())d.push(e.value);b.body=d}return D({originalRequest:{input:a,
+init:b},ampComponent:g})};function eb(a,b){if("."==b)return a;b=b.split(".");for(var c=0;c<b.length;c++){var d=b[c];if(d&&a&&void 0!==a[d]&&(null==a||"object"!=typeof a?0:Object.prototype.hasOwnProperty.call(a,d)))a=a[d];else{a=void 0;break}}return a};function fb(a,b,c,d,e){var g=".";g=void 0===g?".":g;c=void 0===c?0:c;d=void 0===d?!1:d;Va(b.getAttribute("src"),b);var h=M(a.win,"batched-xhr");return gb(b,c,d).then(function(a){void 0!==e&&(a.fetchOpt.method="POST",a.fetchOpt.headers={"Content-Type":"application/x-www-form-urlencoded"},a.fetchOpt.body={ampViewerAuthToken:e});return h.fetchJson(a.xhrUrl,a.fetchOpt)}).then(function(a){return a.json()}).then(function(a){if(null==a)throw Error("Response is undefined.");return eb(a,g||".")})}
+function gb(a,b,c){var d=a.getAttribute("src"),e=za(a,"url-replace");return(1<=b?e.expandUrlAsync(d):Promise.resolve(d)).then(function(d){if(1==b){var g=e.collectUnwhitelistedVarsSync(a);if(0<g.length)throw A().createError("URL variable substitutions in CORS fetches from dynamic URLs (e.g. via amp-bind) require opt-in. "+('Please add data-amp-replace="'+g.join(" ")+'" to the ')+("<"+a.tagName+"> element. See https://bit.ly/amp-var-subs."));}var f={};a.hasAttribute("credentials")&&(f.credentials=a.getAttribute("credentials"));
+c&&(f.cache="reload");return{xhrUrl:d,fetchOpt:f}})};var U;function hb(a,b){var c=void 0,d=a,e=b;var g=function(a){try{return e(a)}catch(l){throw self.reportError(l),l;}};var h=ib(),f=!1;c&&(f=c.capture);d.addEventListener("click",g,h?c:f)}function ib(){if(void 0!==U)return U;U=!1;try{var a={get capture(){U=!0}};self.addEventListener("test-options",null,a);self.removeEventListener("test-options",null,a)}catch(b){}return U};function jb(a,b,c,d){var e={detail:c};Object.assign(e,d);if("function"==typeof a.CustomEvent)return new a.CustomEvent(b,e);a=a.document.createEvent("CustomEvent");a.initCustomEvent(b,!!e.bubbles,!!e.cancelable,c);return a}function kb(a,b){hb(a,b)};var lb=function(){return function e(b,c,d){function g(f,l){if(!c[f]){if(!b[f]){var k="function"==typeof require&&require;if(!l&&k)return k(f,!0);if(h)return h(f,!0);var m=Error("Cannot find module '"+f+"'");throw m.code="MODULE_NOT_FOUND",m;}var y=c[f]={exports:{}};b[f][0].call(y.exports,function(c){return g(b[f][1][c]||c)},y,y.exports,e,b,c,d)}return c[f].exports}for(var h="function"==typeof require&&require,f=0;f<d.length;f++)g(d[f]);return g}({1:[function(b,c){function d(b,c){!function(b,c){if(!b)throw Error("set-dom: "+
+c);}(b&&b.nodeType,"You must provide a valid node to update.");b.nodeType===y&&(b=b.documentElement);c.nodeType===p?g(b,c):e(b,"string"==typeof c?k(c,b.nodeName):c);b[r]||(b[r]=!0,f(b,"mount"))}function e(b,c){if(b.nodeType===c.nodeType)if(b.nodeType===m){if(!function(b,c){return null!=b.getAttribute(d.IGNORE)&&null!=c.getAttribute(d.IGNORE)||(b.getAttribute(d.CHECKSUM)||NaN)===(c.getAttribute(d.CHECKSUM)||NaN)||b.isEqualNode(c)}(b,c))if(g(b,c),b.nodeName===c.nodeName)!function(b,c){var d,e;for(d=
+b.length;d--;){var f=b[d];var g=f.namespaceURI;var h=f.localName;(e=c.getNamedItemNS(g,h))||b.removeNamedItemNS(g,h)}for(d=c.length;d--;)f=c[d],g=f.namespaceURI,h=f.localName,(e=b.getNamedItemNS(g,h))?e.value!==f.value&&(e.value=f.value):(c.removeNamedItemNS(g,h),b.setNamedItemNS(f))}(b.attributes,c.attributes);else{for(c=c.cloneNode();b.firstChild;)c.appendChild(b.firstChild);b.parentNode.replaceChild(c,b)}}else b.nodeValue!==c.nodeValue&&(b.nodeValue=c.nodeValue);else b.parentNode.replaceChild(c,
+f(b,"dismount")),f(c,"mount")}function g(b,c){for(var d,g,k,S,l,p=b.firstChild,m=c.firstChild,T=0;p;)T++,g=h(d=p),p=p.nextSibling,g&&(l||(l={}),l[g]=d);for(p=b.firstChild;m;)T--,c=m,m=m.nextSibling,l&&(k=h(c))&&(S=l[k])?(delete l[k],S!==p?b.insertBefore(S,p):p=p.nextSibling,e(S,c)):p?(d=p,p=p.nextSibling,h(d)?(b.insertBefore(c,d),f(c,"mount")):e(d,c)):(b.appendChild(c),f(c,"mount"));for(g in l)T--,b.removeChild(f(l[g],"dismount"));for(;0<=--T;)b.removeChild(f(b.lastChild,"dismount"))}function h(b){if(b.nodeType===
+m)return(b=b.getAttribute(d.KEY)||b.id)?l+b:void 0}function f(b,c){if(h(b)){var d=document.createEvent("Event"),e={value:b};d.initEvent(c,!1,!1);Object.defineProperty(d,"target",e);Object.defineProperty(d,"srcElement",e);b.dispatchEvent(d)}for(d=b.firstChild;d;)d=f(d,c).nextSibling;return b}d.KEY="data-key";d.IGNORE="data-ignore";d.CHECKSUM="data-checksum";var k=b(2),l="_set-dom-",r=l+"mounted",m=1,y=9,p=11;c.exports=d},{2:2}],2:[function(b,c){function d(b,c){if(c===g){if(h)return r.innerHTML=b,r;
+var d=b.match(y);if(d){c=d[2];d=d.index+d[1].length;var k=d+c.length;b=b.slice(0,d)+b.slice(k);m.innerHTML=c}b=e.parseFromString(b,f);for(c=b.body;m.firstChild;)c.appendChild(m.firstChild);return b.documentElement}return m.innerHTML=b,m.firstChild}var e=window.DOMParser&&new window.DOMParser,g="HTML",h=b=!1,f="application/xhtml+xml";try{var k=e.parseFromString('<wbr class="A"/>',"text/html").body.firstChild,l=document.createElement("div");if(l.appendChild(k),"A"!==l.firstChild.classList[0])throw Error();
+b=!0}catch(p){}k=document.implementation.createHTMLDocument("");var r=k.documentElement,m=k.body;try{r.innerHTML+="",h=!0}catch(p){e.parseFromString('<wbr class="A"/>',f);var y=/(<body[^>]*>)([\s\S]*)<\/body>/}c.exports=b?function(b,c){var f=e.parseFromString(b,"text/html");return f.body?c===g?f.documentElement:f.body.firstChild:d(b,c)}:d},{}]},{},[1])(1)}();function V(a){var b=AMP.BaseElement.call(this,a)||this;b.h=null;b.F=null;b.V=!1;b.X=new N(b.win,function(){return mb(b)});b.A=null;b.Y=null;b.m=M(b.win,"templates");b.L=!1;b.W=null;b.B=null;b.o=!1;b.O=null;b.w=null;b.R=!1;b.$=null;b.registerAction("refresh",function(){if(b.L)return nb(b),W(b,!1,!0)},100);b.registerAction("changeToLayoutContainer",function(){return ob(b)},100);b.I=null;return b}var X=AMP.BaseElement;V.prototype=aa(X.prototype);V.prototype.constructor=V;
+if(da)da(V,X);else for(var Y in X)if("prototype"!=Y)if(Object.defineProperties){var pb=Object.getOwnPropertyDescriptor(X,Y);pb&&Object.defineProperty(V,Y,pb)}else V[Y]=X[Y];V.fa=X.prototype;n=V.prototype;n.isLayoutSupported=function(a){return"fixed"==a||"fixed-height"==a||"responsive"==a||"fill"==a||"flex-item"==a||"fluid"==a||"intrinsic"==a};
+n.buildCallback=function(){var a=this;this.F=this.getViewport();var b=Ba(this.getAmpDoc());this.I=new R(b,this.m);this.o=this.element.hasAttribute("load-more");this.W=this.element.getAttribute("src");this.h=qb(this);this.element.appendChild(this.h);this.element.hasAttribute("aria-live")||this.element.setAttribute("aria-live","polite");this.element.hasAttribute("auto-resize")&&A().warn("amp-list","auto-resize attribute is deprecated and its behavior is disabled. This feature will be relaunched under a new name soon. Please see https://github.com/ampproject/amphtml/issues/18849");
+Ja(this.element).then(function(b){a.B=b})};n.reconstructWhenReparented=function(){return!1};n.layoutCallback=function(){var a=this;this.L=!0;var b=this.getPlaceholder();b&&rb(this,b);this.F.onResize(function(){sb(a)});this.o&&tb(this);return W(this)};
+function tb(a){a.mutateElement(function(){Z(a).initializeLoadMore();var b=a.getOverflowElement();b&&la(b);a.element.warnOnMissingOverflow=!1}).then(function(){ub(a);kb(Z(a).getLoadMoreFailedClickable(),function(){return vb(a,!0)});kb(Z(a).getLoadMoreButtonClickable(),function(){return vb(a)})})}function sb(a){a.o?wb(a,a.h):rb(a,a.h)}function Z(a){a.O||(a.O=new va(a.element));return a.O}
+function ub(a){var b,c;a.measureMutateElement(function(){b=Z(a).getLoadMoreButton().offsetHeight;c=a.element.offsetHeight},function(){H(a.h,{"max-height":"calc(100% - "+(b+"px")+")"});a.element.changeSize(c+b)})}
+n.mutatedAttributesCallback=function(a){function b(a){c.element.setAttribute("src","");a=x(a)?a:[a];nb(c,!1);return xb(c,a,!1)}var c=this;B().info("amp-list","mutate:",this.element,a);var d=a.src,e=a.state;if(void 0!==d)if("string"===typeof d){if(this.L){nb(this);var g=W(this)}}else"object"===typeof d?g=b(d):this.user().error("amp-list",'Unexpected "src" type: '+d);else void 0!==e&&(A().error("amp-list","[state] is deprecated, please use [src] instead."),g=b(e));var h=a["is-layout-container"];h&&
+ob(this);if(ha().test)return g};n.isLoadingReused=function(){return this.element.hasAttribute("reset-on-refresh")};function qb(a){var b=a.win.document.createElement("div");b.setAttribute("role","list");a.o||a.applyFillContent(b,!0);return b}function yb(a,b){a.forEach(function(a){a.hasAttribute("role")||a.setAttribute("role","listitem");b.appendChild(a)})}function zb(a,b){if(b||a.V)a.toggleFallback(b),a.V=b}
+function nb(a,b){if((b=void 0===b?!0:b)&&a.element.hasAttribute("reset-on-refresh")||"always"===a.element.getAttribute("reset-on-refresh"))a.togglePlaceholder(!0),a.toggleLoading(!0,!0),a.mutateElement(function(){zb(a,!1);if(a.B){var b=(b=a.h.children)?Array.prototype.slice.call(b):[];a.B.rescan([],b,{fast:!0,update:!1})}ra(a.h)})}
+function W(a,b,c){b=void 0===b?!1:b;c=void 0===c?!1:c;if(!a.element.getAttribute("src"))return Promise.resolve();if(a.I.isSupported())c=Ab(a,c);else{var d=a.element.getAttribute("items")||"items";c=Bb(a,c).then(function(c){var e=c;"."!=d&&(e=eb(c,d));C("undefined"!==typeof e,'Response must contain an array or object at "%s". %s',d,a.element);a.element.hasAttribute("single-item")&&!x(e)&&(e=[e]);e=A().assertArray(e);if(a.element.hasAttribute("max-items")){var h=e,f=parseInt(a.element.getAttribute("max-items"),
+10);f<h.length&&(h=h.slice(0,f));e=h}a.o&&(h=a.element.getAttribute("load-more-bookmark")||"load-more-src",a.w=eb(c,h));return xb(a,e,b,c)})}return c.catch(function(c){var d=c?jb(a.win,"amp-list.error",D({response:c.response})):null,e=za(a.element,"action");e.trigger(a.element,"fetch-error",d,1);if(b)throw c;a.toggleLoading(!1);if(a.getFallback())zb(a,!0),a.togglePlaceholder(!1);else throw c;})}
+function Ab(a,b){var c;return gb(a.element,Cb(a),b).then(function(b){var d=c=b,g=c.xhrUrl;if(!1!==c.fetchOpt.ampCors){var h=a.win,f=Q(g);f=v(f.search);C(!("__amp_source_origin"in f),"Source origin is not allowed in %s",g);h=Q(Wa(h.location.href)).origin;h=encodeURIComponent("__amp_source_origin")+"="+encodeURIComponent(h);h&&(g=g.split("#",2),f=g[0].split("?",2),h=f[0]+(f[1]?"?"+f[1]+"&"+h:"?"+h),g=h+=g[1]?"#"+g[1]:"")}d.xhrUrl=g;d=c;h=c.xhrUrl;g=(g=c.fetchOpt)||{};f=a.win;f=f.origin||Q(f.location.href).origin;
+h=Q(h).origin;f==h&&(g.headers=g.headers||{},g.headers["AMP-Same-Origin"]="true");d.fetchOpt=g;bb(b.fetchOpt);b=D({ampListAttributes:{items:a.element.getAttribute("items")||"items",singleItem:a.element.hasAttribute("single-item"),maxItems:a.element.getAttribute("max-items")}});return a.I.fetchAndRenderTemplate(a.element,c,null,b)}).then(function(a){C(a&&"string"===typeof a.html,"Expected response with format {html: <string>}. Received: ",a);c.fetchOpt.responseType="application/json";return a},function(a){throw A().createError("Error proxying amp-list templates",
+a);}).then(function(b){return xb(a,b,!1)})}function xb(a,b,c,d){var e=new ea,g=e.promise,h=e.resolve,f=e.reject;a.A||a.X.schedule();a.A={data:b,resolver:h,rejecter:f,append:c,payload:d};a.Y&&c&&(a.A.payload=d||{});return g}
+function mb(a){function b(){d();e.rejecter()}function c(){d();e.resolver()}function d(){a.A!==e?a.X.schedule(1):(a.Y=a.A.data,a.A=null)}var e=a.A,g=a.I.isSupported(),h=a.I.renderTemplate(a.element,e.data).then(function(b){return Db(a,b,e.append)}).then(function(b){return Eb(a,b,e.append)});if(!g){var f=e.payload;h=h.then(function(){if(a.o){var b=[];b.push(Fb(a,Z(a).getLoadMoreButton(),f));b.push(Fb(a,Z(a).getLoadMoreEndElement(),f));b=Promise.all(b)}else b=Promise.resolve();return b}).then(function(){return a.o?
+Gb(a):Promise.resolve()})}h.then(c,b)}function Fb(a,b,c){return b&&a.m.hasTemplate(b)?a.m.findAndRenderTemplate(b,c).then(function(c){return a.mutateElement(function(){ra(b);b.appendChild(c)})}):Promise.resolve()}
+function Db(a,b,c){function d(b){var d=c?[]:[a.h];return b.rescan(e,d,{fast:!0,update:!0}).then(function(){return e},function(){return e})}var e=x(b)?b:[b],g=a.element.getAttribute("binding");if("no"===g)return Promise.resolve(e);var h=e.some(function(a){return a.hasAttribute("i-amphtml-binding")||!!a.querySelector("[i-amphtml-binding]")});if(!h)return Promise.resolve(e);if("refresh"===g){if(a.B&&a.B.signals().get("FIRST_MUTATE"))return d(a.B);Ja(a.element).then(function(a){a&&a.rescan(e,[],{fast:!0,
+update:!1})});return Promise.resolve(e)}return Ja(a.element).then(function(a){return a?d(a):Promise.resolve(e)})}
+function Eb(a,b,c){c=void 0===c?!1:c;B().info("amp-list","render:",a.element,b);var d=a.h;return a.mutateElement(function(){a.toggleLoading(!1);a.getFallback()&&zb(a,!1);a.togglePlaceholder(!1);var e=a.win;if(e.__AMP__EXPERIMENT_TOGGLES)var g=e.__AMP__EXPERIMENT_TOGGLES;else{e.__AMP__EXPERIMENT_TOGGLES=Object.create(null);g=e.__AMP__EXPERIMENT_TOGGLES;if(e.AMP_CONFIG)for(var h in e.AMP_CONFIG){var f=e.AMP_CONFIG[h];"number"===typeof f&&0<=f&&1>=f&&(g[h]=Math.random()<f)}if(e.AMP_CONFIG&&Array.isArray(e.AMP_CONFIG["allow-doc-opt-in"])&&
+0<e.AMP_CONFIG["allow-doc-opt-in"].length&&(h=e.AMP_CONFIG["allow-doc-opt-in"],f=e.document.head.querySelector('meta[name="amp-experiments-opt-in"]'))){f=f.getAttribute("content").split(",");for(var k=0;k<f.length;k++)-1!=h.indexOf(f[k])&&(g[f[k]]=!0)}Object.assign(g,Xa(e));if(e.AMP_CONFIG&&Array.isArray(e.AMP_CONFIG["allow-url-opt-in"])&&0<e.AMP_CONFIG["allow-url-opt-in"].length)for(h=e.AMP_CONFIG["allow-url-opt-in"],e=v(e.location.originalHash||e.location.hash),f=0;f<h.length;f++)k=e["e-"+h[f]],
+"1"==k&&(g[h[f]]=!0),"0"==k&&(g[h[f]]=!1)}var l=!!g["amp-list-diffing"];if(l&&d.hasChildNodes()){var r=qb(a);yb(b,r);var m=lb.default||lb;m.KEY="i-amphtml-key";m(d,r)}else c||ra(d),yb(b,d);g=jb(a.win,"amp:dom-update",null,{bubbles:!0});a.h.dispatchEvent(g);a.element.getResources().getResourceForElement(a.element).resetPendingChangeSize();sb(a)})}
+function rb(a,b){"container"!=a.element.getAttribute("layout")&&a.measureElement(function(){var c=b.scrollHeight;c>a.element.offsetHeight&&a.attemptChangeHeight(c).catch(function(){})})}function wb(a,b){var c=a.w?Z(a).getLoadMoreButton():Z(a).getLoadMoreEndElement();Hb(a,c,b)}
+function Hb(a,b,c){"container"!=a.element.getAttribute("layout")&&a.measureElement(function(){var d=c.scrollHeight,e=b?b.offsetHeight:0;d+e>a.element.offsetHeight&&a.attemptChangeHeight(d+e).then(function(){a.R=!1;"auto"===a.element.getAttribute("load-more")&&Ib(a);H(a.h,{"max-height":""})}).catch(function(){a.R=!0;ub(a)})})}
+function ob(a){var b=a.element.getAttribute("layout");return"container"==b?Promise.resolve():a.mutateElement(function(){switch(b){case "responsive":a.element.classList.remove("i-amphtml-layout-responsive");H(a.element,{height:"",width:""});break;case "flex-item":H(a.element,{height:"",width:""});break;case "fixed":a.element.classList.remove("i-amphtml-layout-fixed");H(a.element,{height:""});break;case "fixed-height":a.element.classList.remove("i-amphtml-layout-fixed-height");H(a.element,{height:"",
+width:""});break;case "intrinsic":a.element.classList.remove("i-amphtml-layout-intrinsic")}a.element.changeSize();a.element.classList.remove("i-amphtml-layout-size-defined");a.h.classList.remove("i-amphtml-fill-content","i-amphtml-replaced-content");var c=a.getOverflowElement();c&&la(c);a.element.classList.add("i-amphtml-layout-container");a.element.setAttribute("layout","container")})}
+function Gb(a){if(a.w){var b="auto"===a.element.getAttribute("load-more");b&&Jb(a);return a.mutateElement(function(){Z(a).toggleLoadMoreLoading(!1);H(Z(a).getLoadMoreButton(),{visibility:""})})}return a.mutateElement(function(){return Z(a).setLoadMoreEnded()})}
+function vb(a,b){b=void 0===b?!1:b;if(a.w)a.element.setAttribute("src",a.w),a.w=null;else if(!b)return Promise.resolve();a.mutateElement(function(){Z(a).toggleLoadMoreLoading(!0)});return W(a,!0).then(function(){return a.mutateElement(function(){a.w?Z(a).toggleLoadMoreLoading(!1):Z(a).setLoadMoreEnded()})}).then(function(){wb(a,a.h)}).catch(function(){Kb(a)})}function Kb(a){a.mutateElement(function(){return Z(a).setLoadMoreFailed()}).then(function(){Hb(a,Z(a).getLoadMoreFailedElement(),a.h)})}
+function Jb(a){a.$||(a.$=a.F.onChanged(function(){return Ib(a)}))}function Ib(a){if(!a.R){var b=a.h.lastChild||a.h;a.F.getClientRectAsync(b).then(function(b){var c=a.F.getHeight(),e=a.F.getScrollTop();if(e+3*c>b.bottom)return vb(a)})}}function Bb(a,b){b=void 0===b?!1:b;return cb(a.element).then(function(c){var d=b;d=void 0===d?!1:d;return fb(a.getAmpDoc(),a.element,Cb(a),d,c)})}
+function Cb(a){var b=a.element.getAttribute("src"),c=1;if(b==a.W||Q(Wa(b)).origin==Q(Wa(a.getAmpDoc().win.location)).origin)c=2;return c}(function(a){a.registerElement("amp-list",V,"amp-list[load-more] [load-more-button].amp-visible,amp-list[load-more] [load-more-end].amp-visible,amp-list[load-more] [load-more-failed].amp-visible,amp-list[load-more] [load-more-loading].amp-visible{display:block}amp-list[load-more] [load-more-button].i-amphtml-default-ui,amp-list[load-more] [load-more-failed].i-amphtml-default-ui,amp-list[load-more] [load-more-loading].i-amphtml-default-ui{height:80px;padding:12px 0px;box-sizing:border-box}.i-amphtml-list-load-more-button,amp-list[load-more] [load-more-button].i-amphtml-default-ui,amp-list[load-more] [load-more-failed].i-amphtml-default-ui,amp-list[load-more] [load-more-loading].i-amphtml-default-ui{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-weight:700;font-size:14px;text-transform:uppercase;letter-spacing:.1em;color:#333;text-align:center}amp-list[load-more] [load-more-loading].i-amphtml-default-ui .i-amphtml-list-load-more-spinner{display:inline-block;width:40px;height:40px;margin:8px 0px;background-image:url(\"data:image/svg+xml;charset=utf-8,%3Csvg viewBox='0 0 40 40' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3ClinearGradient id='a'%3E%3Cstop stop-color='%23333' stop-opacity='.75'/%3E%3Cstop offset='100%25' stop-color='%23333' stop-opacity='0'/%3E%3C/linearGradient%3E%3C/defs%3E%3Cpath d='M11 4.4A18 18 0 1 0 38 20' fill='none' stroke='url(%23a)' stroke-width='1.725'/%3E%3C/svg%3E\");animation:amp-list-load-more-spinner 1000ms linear infinite}@keyframes amp-list-load-more-spinner{0%{transform:rotate(0deg)}to{transform:rotate(360deg)}}.i-amphtml-list-load-more-button{border:none;display:inline-block;background-color:rgba(51,51,51,0.75);color:#fff;margin:4px 0px;padding:0px 32px;box-sizing:border-box;height:48px;border-radius:24px}[load-more] div[role=list]{overflow-y:hidden}.i-amphtml-list-load-more-button,.i-amphtml-list-load-more-button label,.i-amphtml-list-load-more-icon{cursor:pointer}.i-amphtml-list-load-more-button:hover{background-color:#333}.i-amphtml-list-load-more-button.i-amphtml-list-load-more-button-small{margin:0px;padding:4px 16px;height:32px}.i-amphtml-list-load-more-button label{display:inline-block;vertical-align:middle;line-height:24px}amp-list[load-more] [load-more-failed].i-amphtml-default-ui .i-amphtml-list-load-more-message{line-height:24px}amp-list[load-more] [load-more-failed].i-amphtml-default-ui .i-amphtml-list-load-more-icon{height:24px;width:24px;display:inline-block;vertical-align:middle;background-image:url(\"data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24'%3E%3Cpath fill='%23fff' d='M17.65 6.35A7.958 7.958 0 0 0 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08A5.99 5.99 0 0 1 12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z'/%3E%3C/svg%3E\")}\n/*# sourceURL=/extensions/amp-list/0.1/amp-list.css*/")})(self.AMP);
 })});
 
 //# sourceMappingURL=amp-list-0.1.js.map
```

Reply:
Bumping this thread. Let me know if there's any reason this can't be merged today.

Reply:
Ah, it seems Closure can recognize UMD wrappers.